streams/src/main/java/org/apache/kafka/streams/kstream/Consumed.java

```diff
@@ -55,9 +56,12 @@ public class Consumed<K, V> implements NamedOperation<Consumed<K, V>> {
     protected Serde<K> keySerde;
     protected Serde<V> valueSerde;
     protected TimestampExtractor timestampExtractor;
-    protected Topology.AutoOffsetReset resetPolicy;
+    protected AutoOffsetReset resetPolicy;
+    @Deprecated
+    protected Topology.AutoOffsetReset resetPolicy; // Replaced with new AutoOffsetReset class introduced in 4.0. 
```

Comment:
```suggestion
    protected Topology.AutoOffsetReset legacyResetPolicy;
```

Reply:
We cannot have two member variable `resetPolicy` -- lets rename the old one in a meaningful way -- this also allows us to avoid the comment.

Reply:
Updated this